base/database.py

```diff
@@ -3560,12 +3560,12 @@ def id(cls, ea):
 
             res = type.flags(ea, idaapi.DT_TYPE)
             if res != FF_STRUCT:
-                raise E.MissingTypeOrAttribute(u"{:s}.id({:#x}) : The type at specified address is not an FF_STRUCT({:#x}) and is instead {:#x}.".format('.'.join((__name__, 'type', 'structure')), ea, FF_STRUCT, res))
+                raise E.MissingTypeOrAttribute(u"{:s}.id({:#x}) : The type at specified address is not an FF_STRUCT({:#x}) and is instead {:#x}.".format('.'.join([__name__, 'type', 'structure']), ea, FF_STRUCT, res))
```

Comment:
This might as well use `cls.__name__` instead of hardcoding the string "structure".